flask_security/decorators.py

```diff
@@ -53,7 +53,11 @@ def _get_unauthorized_view():
             except BuildError:
                 view = None
         utils.do_flash(*utils.get_message('UNAUTHORIZED'))
-        return redirect(view or request.referrer or '/')
+        redirect_to = '/'
+        if request.referrer and not request.referrer.endswith(request.path):
```

Comment:
I agree with the idea of checking the `request.referrer` and `request.path` overlap. I am not sure if it work well if the referrer contains a query arguments.

a. we should check overlap with `request.full_path`
b. compare parsed and normalized URLs (with/without host, relative/absolute path) ...